testing/integration/tests/fil-ipld-actor/src/actor.rs

```diff
@@ -118,15 +150,111 @@ fn test_read_block() {
         );
 
         // Test an offset that overflows an i32:
-        sdk::sys::ipld::block_read(id, (i32::MAX as u32) + 1, buf.as_mut_ptr(), 0)
-            .expect_err("expected it to fail");
+        let res = sdk::sys::ipld::block_read(id, (i32::MAX as u32) + 1, buf.as_mut_ptr(), 0);
+        assert_eq!(res, Err(ErrorNumber::IllegalArgument));
 
         // Test a length that overflows an i32
-        sdk::sys::ipld::block_read(id, 0, buf.as_mut_ptr(), (i32::MAX as u32) + 1)
-            .expect_err("expected it to fail");
+        let res = sdk::sys::ipld::block_read(id, 0, buf.as_mut_ptr(), (i32::MAX as u32) + 1);
+        assert_eq!(res, Err(ErrorNumber::IllegalArgument));
 
         // Test a combined length + offset that overflow an i32
-        sdk::sys::ipld::block_read(id, (i32::MAX as u32) - 10, buf.as_mut_ptr(), 20)
-            .expect_err("expected it to fail");
+        let res = sdk::sys::ipld::block_read(id, (i32::MAX as u32) - 10, buf.as_mut_ptr(), 20);
+        assert_eq!(res, Err(ErrorNumber::IllegalArgument));
+    }
+}
+
+fn test_create_block() {
+    unsafe {
+        // Test creating a block with invalid codec
+        let test_bytes = gen_test_bytes(100);
+        let invalid_codec_id = 191919;
+        let res = sdk::sys::ipld::block_create(
+            invalid_codec_id,
+            test_bytes.as_ptr(),
+            test_bytes.len() as u32,
+        );
+        assert_eq!(res, Err(ErrorNumber::IllegalCodec));
+
+        // Creating a block just within the 1Mib limit should work
+        let test_bytes = gen_test_bytes((1 << 20) - 8);
+        sdk::sys::ipld::block_create(DAG_CBOR, test_bytes.as_ptr(), test_bytes.len() as u32)
+            .expect("should be within block limit");
+
+        // Test creating a too large block
+        let test_bytes = gen_test_bytes((1 << 20) + 8);
+        let res =
+            sdk::sys::ipld::block_create(DAG_CBOR, test_bytes.as_ptr(), test_bytes.len() as u32);
+        assert_eq!(res, Err(ErrorNumber::LimitExceeded));
+
+        // TODO (fridrik): Test for TooManyBlock error requires INT32_MAX blocks
```

Comment:
I wouldn't bother with this one, honestly. Given the current gas model, there's no way to hit this limit at the moment.